Formula/folly.rb

```diff
@@ -41,21 +39,19 @@ def install
     ENV.cxx11
 
     cd "folly" do
-      # Workaround for "no matching function for call to 'clock_gettime'"
-      # See upstream PR from 2 Oct 2016 facebook/folly#488
-      if DevelopmentTools.clang_build_version >= 800
+      if MacOS.version == "10.11" && MacOS::Xcode.installed? && MacOS::Xcode.version >= "8.0"
+        # Workaround for "no matching function for call to 'clock_gettime'"
+        # See upstream PR from 2 Oct 2016 facebook/folly#488
         inreplace ["Benchmark.cpp", "Benchmark.h"] do |s|
-          s.gsub! "detail::DEFAULT_CLOCK_ID",
-                  "(clockid_t)detail::DEFAULT_CLOCK_ID"
           s.gsub! "clock_gettime(CLOCK_REALTIME",
-                  "clock_gettime((clockid_t)CLOCK_REALTIME", false
+                  "clock_gettime((clockid_t)CLOCK_REALTIME"
+          s.gsub! "clock_getres(CLOCK_REALTIME",
+                  "clock_getres((clockid_t)CLOCK_REALTIME", false
         end
-      end
 
-      # Fix "candidate function not viable: no known conversion from
-      # 'folly::detail::Clock' to 'clockid_t' for 1st argument"
-      # See upstream PR mentioned above
-      if MacOS.version == "10.11" && MacOS::Xcode.installed? && MacOS::Xcode.version >= "8.0"
+        # Fix "candidate function not viable: no known conversion from
+        # 'folly::detail::Clock' to 'clockid_t' for 1st argument"
+        # See upstream PR mentioned above
         inreplace "portability/Time.h", "typedef uint8_t clockid_t;", ""
```

Comment:
I'm still not seeing why the addition of the casts is required. The casts aren't required on 10.12.

If I'm understanding everything correctly (I don't have easy access to a 10.11 machine to test with), Folly is currently defining its implementation of `clock_gettime` on OSX, but the declaration of `clockid_t` isn't compatible with the way OSX's headers define it. That means the only thing that needs to be done for OSX 10.11 to work is to make the declaration of `clockid_t` conditional on it not already being declared, correct?

As a side note, as-of https://github.com/facebook/folly/commit/823ecf8917d7f0534ed02b8d38bfc966cb0fd7b6, Folly can be compiled (properly) against Boost >= 1.61.